__tests__/setup-go.test.ts

```diff
@@ -322,6 +325,100 @@ describe('setup-go', () => {
+    message: string;
+  }
+
+  //
+  // problem matcher regex pattern tests
+
+  function testMatch(line: string): Annotation {
+    let annotation = <Annotation>{};
+
+    let match = matcherRegExp.exec(line);
+    if (match) {
+      annotation.line = parseInt(match[matcherPattern.line], 10);
+      annotation.column = parseInt(match[matcherPattern.column], 10);
+      annotation.file = match[matcherPattern.file].trim();
+      annotation.message = match[matcherPattern.message].trim();
+    }
+
+    return annotation;
+  }
+
+  it('matches on relative unix path', async () => {
+    let line = './main.go:13:2: undefined: fmt.Printl';
+    let annotation = testMatch(line);
+    expect(annotation).toBeDefined();
+    expect(annotation.line).toBe(13);
+    expect(annotation.column).toBe(2);
+    expect(annotation.file).toBe('./main.go');
+    expect(annotation.message).toBe('undefined: fmt.Printl');
+  });
+
+  it('matches on unix path up the tree', async () => {
+    let line = '../main.go:13:2: undefined: fmt.Printl';
+    let annotation = testMatch(line);
+    expect(annotation).toBeDefined();
+    expect(annotation.line).toBe(13);
+    expect(annotation.column).toBe(2);
+    expect(annotation.file).toBe('../main.go');
+    expect(annotation.message).toBe('undefined: fmt.Printl');
+  });
+
+  it('matches on rooted unix path', async () => {
+    let line = '/assert.go:4:1: missing return at end of function';
+    let annotation = testMatch(line);
+    expect(annotation).toBeDefined();
+    expect(annotation.line).toBe(4);
+    expect(annotation.column).toBe(1);
+    expect(annotation.file).toBe('/assert.go');
+    expect(annotation.message).toBe('missing return at end of function');
+  });
+
+  it('matches on unix path with whitespace', async () => {
+    let line = '   ./assert.go:5:2: missing return at end of function   ';
+    let annotation = testMatch(line);
+    expect(annotation).toBeDefined();
+    expect(annotation.line).toBe(5);
+    expect(annotation.column).toBe(2);
+    expect(annotation.file).toBe('./assert.go');
+    expect(annotation.message).toBe('missing return at end of function');
+  });
+
+  it('matches on relative windows path', async () => {
+    let line = '.\\main.go:13:2: undefined: fmt.Printl';
+    let annotation = testMatch(line);
+    expect(annotation).toBeDefined();
+    expect(annotation.line).toBe(13);
+    expect(annotation.column).toBe(2);
+    expect(annotation.file).toBe('.\\main.go');
+    expect(annotation.message).toBe('undefined: fmt.Printl');
+  });
+
+  it('matches on windows path up the tree', async () => {
+    let line = '..\\main.go:13:2: undefined: fmt.Printl';
+    let annotation = testMatch(line);
+    expect(annotation).toBeDefined();
+    expect(annotation.line).toBe(13);
+    expect(annotation.column).toBe(2);
+    expect(annotation.file).toBe('..\\main.go');
+    expect(annotation.message).toBe('undefined: fmt.Printl');
+  });
+
+  it('only matches leading dots on unix path', async () => {
```

Comment:
The description doesnt align with the behavior

Should the description be updated, or should the behavior be fixed?

Reply:
I got confused why a match was returned at all.

The regexp doesn't have `^$` around the pattern. Should it?

Reply:
I think it matches??  In this line:
```
x./assert.go:5:2
```
it's validating that it only matches the leading dot(s) and not the x.  Not sure this output would ever happen but my initial regex had a bug where I wasn't escaping the dot so it was match 0-2 of any char in from of the slash which is not correct.

Do you have a better suggestion for the title?

Reply:
fixed.